src/clients/near.ts

```diff
@@ -564,4 +569,176 @@ export class NearBridgeClient {
+   * This function orchestrates the entire fast transfer process:
+   * 1. Fetches and parses the InitTransfer event from the EVM transaction
+   * 2. Gets the corresponding NEAR token ID using getBridgedToken
+   * 3. Executes the fast finalize transfer on NEAR
+   *
+   * @param originChain - The EVM chain where the original transfer was initiated
+   * @param evmTxHash - Transaction hash of the InitTransfer on the EVM chain
+   * @param evmClient - EVM bridge client for parsing the transaction
+   * @param storageDepositAmount - Optional storage deposit amount in yoctoNEAR
+   * @returns Promise resolving to the NEAR transaction hash
+   * @throws {Error} If the origin chain is not supported for fast transfers
+   * @throws {Error} If the EVM transaction or event cannot be found/parsed
+   * @throws {Error} If the fast transfer execution fails
+   */
+  async nearFastTransfer(
+    originChain: ChainKind,
+    evmTxHash: string,
+    evmClient: EvmBridgeClient,
+    storageDepositAmount?: string,
+  ): Promise<string> {
+    // Validate supported chains for fast transfer
+    if (!isEvmChain(originChain)) {
+      throw new Error(`Fast transfer is not supported for chain kind: ${ChainKind[originChain]}`)
+    }
+
+    // Step 1: Parse the InitTransfer event from EVM transaction
+    const transferEvent = await evmClient.getInitTransferEvent(evmTxHash)
+
+    // Step 2: Get the NEAR token ID for the EVM token using getBridgedToken
+    const omniTokenAddress = omniAddress(originChain, transferEvent.tokenAddress)
+    const nearTokenAddress = await getBridgedToken(omniTokenAddress, ChainKind.Near)
+
+    if (!nearTokenAddress) {
+      throw new Error(`No bridged token found on NEAR for ${omniTokenAddress}`)
+    }
+
+    const nearTokenId = nearTokenAddress.split(":")[1] // Extract account ID from near:account.near
+
+    // Step 3: Normalize the amount & fee based on source/destination decimals
+    const sourceDecimals = await getTokenDecimals(addresses.near, omniTokenAddress)
+    const destinationDecimals = await getTokenDecimals(addresses.near, nearTokenAddress)
+    const normalizedAmount = normalizeAmount(
+      transferEvent.amount,
+      sourceDecimals.decimals,
+      destinationDecimals.decimals,
+    )
+    const normalizedFee = normalizeAmount(
+      transferEvent.fee,
+      sourceDecimals.decimals,
+      destinationDecimals.decimals,
+    )
+
+    // Step 4: Construct the transfer ID
+    const transferId: TransferId = {
+      origin_chain: originChain, // Use numeric enum value
+      origin_nonce: transferEvent.originNonce,
+    }
+
+    // Step 5: Execute the fast finalize transfer
+    const fastTransferArgs: FastFinTransferArgs = {
+      token_id: nearTokenId,
+      amount: normalizedAmount.toString(),
+      transfer_id: transferId,
+      recipient: transferEvent.recipient,
+      fee: {
+        fee: normalizedFee.toString(),
+        native_fee: transferEvent.nativeTokenFee.toString(),
```

Comment:
In practice native tokens have the same decimals on both chains but perhaps we should still normalize this as well